ui/components/src/ui-components/filtering/FilteringInput.tsx

```diff
@@ -20,41 +21,83 @@ import { ResultSelector, ResultSelectorProps } from "./ResultSelector";
 interface FilteringInputState {
   /** A string which will be used for search */
   searchText: string;
-  /** @internal */
-  context: InputContext;
+  /**
+   *  Tells the component if the search was started.
+   *  Gets reset to false if search is canceled/cleared or searchText is changed.
+  */
+  searchStarted: boolean;
+  /* Used for resetting the state of [[ResultSelector]] component */
+  resultSelectorKey: number;
+  /**
+   *  Parameter used to remember `props.filteringInProgress`.
+   *  Used in getDerivedStateFromProps to compare previous `props.filteringInProgress`
+   *  with current `props.filteringInProgress`.
+   *  Used for supporting deprecated usage of [[FilteringInput]] component
+   */
+  filteringInProgress?: boolean;
 }
 
 /** [[FilteringInput]] React Component properties
  * @public
+ * @deprecated
  */
-export interface FilteringInputProps extends CommonProps {
+export interface DEPRECATED_FilteringInputProps extends CommonProps {
   /** Filtering should start */
   onFilterStart: (searchText: string) => void;
   /** Filtering is canceled while still in progress */
   onFilterCancel: () => void;
   /** Filtering is cleared after everything's loaded */
   onFilterClear: () => void;
-  /** Tells the component if parent component is still handling the filtering */
+  /**
+   * Tells the component if parent component is still handling the filtering.
+   * @deprecated use `status` to provide information about filtering status instead.
+   */
   filteringInProgress: boolean;
   /** [[ResultSelector]] React Component properties */
   resultSelectorProps?: ResultSelectorProps;
   /** Specify that the <input> element should automatically get focus */
   autoFocus?: boolean;
 }
 
+/**
+ * [[FilteringInput]] React Component properties
+ * @public
+ */
+export interface NEW_FilteringInputProps extends CommonProps {
+  /** Filtering should start */
+  onFilterStart: (searchText: string) => void;
+  /** Filtering is canceled while still in progress */
+  onFilterCancel: () => void;
+  /** Filtering is cleared after everything's loaded */
+  onFilterClear: () => void;
+  /**
+   * Tells the component what is the status of filtering.
+   * @beta
+   */
+  status: FilteringInputStatus;
+  /** [[ResultSelector]] React Component properties */
+  resultSelectorProps?: ResultSelectorProps;
```

Comment:
I think it's worth documenting that this attribute should be memoized as it's used to determine that the search was reset and result selector should clear its internal state.

Reply:
Added some more documentation